server/embed/etcd.go

```diff
@@ -669,17 +692,17 @@ func configureClientListeners(cfg *Config) (sctxs map[string]*serveCtx, err erro
 			sctx.l = transport.LimitListener(sctx.l, int(fdLimit-reservedInternalFDNum))
 		}
 
-		defer func(u url.URL) {
-			if err == nil {
+		defer func(addr string) {
+			if err == nil || sctx.l == nil {
```

Comment:
I think we should pass *sctx* in the defer function instead of address.

https://go.dev/play/p/ZzmQADcBkJl